dojo/engine_tools/helpers.py

```diff
@@ -0,0 +1,202 @@
+    """
+    finding_list = []
+    finding_exclusion_list = []
+    
+    for finding in findings:
+        priority = calculate_vulnerability_priority(finding)
+        
+        finding.priority = priority
+        
+        if priority > 90:
+            finding_exclusion = FindingExclusion.objects.filter(unique_id_from_tool=finding.cve)
+            finding.risk_status = "On Blacklist"
+            finding.tags.add("black_list")
+            
+            if not finding_exclusion.exists():
+                new_finding_exclusion = FindingExclusion(
+                    type="black_list",
+                    unique_id_from_tool=finding.cve,
+                    expiration_date=timezone.now() + timedelta(days=90),
+                    status_updated_at=None,
+                    status_updated_by=None,
+                    reviewed_at=None,
+                    user_history=None,
+                    product=None,
+                    finding=None,
+                    reason="Highly exploitable vulnerability.",
+                    status="Accepted",
+                    final_status="Accepted",
+                    accepted_by=None,
+                    created_by=None
+                )
+                finding_exclusion_list.append(new_finding_exclusion)
+        finding_list.append(finding)
+    
+    Finding.objects.bulk_update(finding_list, ["priority", "risk_status"])     
+    FindingExclusion.objects.bulk_create(finding_exclusion_list)
+            
+    return len(finding_exclusion_list)
+
+
+@app.task
+def check_priorization():
+    # Get all vulnerabilities
+    
+    all_vulnerabilities = Finding.objects.filter(
+        active=True,
+    ).filter(
+        Q(tags__name__icontains="prisma") | Q(tags__name__icontains="tenable")
+    )
+    
+    # Identify critical vulnerabilities
+    blacklist_new_items = identify_critical_vulnerabilities(all_vulnerabilities)
+    
+    return {
+        "message": f"{blacklist_new_items} added to blacklist"
+    }
+    
+
+@app.task
+def check_expiring_findingexclusions():
+    days_before_expiration = 5
+    expiration_threshold = timezone.now() + timedelta(days=days_before_expiration)
+    
+    expiring_objects = FindingExclusion.objects.filter(
+        expiration_date__lte=expiration_threshold,
+        notification_sent=False
+    ).exclude(status__in=['Accepted', 'Rejected'])
+    
+    for fex in expiring_objects:
+        create_notification(
+            event="other",
+            title=f"{fex.unique_id_from_tool} | Whitelist request about to expire.",
+            description=f"The request to whitelist {fex.unique_id_from_tool} expires on {fex.expiration_date.date()}, it is recommended to review the request as soon as possible.",
+            url=reverse("finding_exclusion", args=[str(fex.pk)]),
+        )
+        
+        fex.notification_sent = True
+        fex.save()
+        
+        
+@app.task
+def add_findings_to_whitelist(unique_id_from_tool):
```

Comment:
add a note indicating that it was added by withelist indicating the url of the withelist